src/kernel/stream_io.c

```diff
@@ -0,0 +1,42 @@
+#include <stddef.h>
+#include <kernel/file.h>
+#include <kernel/stream_io.h>
+#include <MK64F12.h>
+size_t kernel_write(int fd, char *buf, size_t bytes) {
+    size_t r = 0;
+    ftab_entry_t *entry = &file_descriptor_table.entries[fd];
+    // FIXME ABSTRACT PID FETCH
```

Comment:
Hmmm

Reply:
hey look, I actually DID abstract that and then forgot to use it...